api/envoy/extensions/filters/http/composite/v3/composite.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.composite.v3;
+
+import "envoy/config/core/v3/extension.proto";
+
+import "google/protobuf/any.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.composite.v3";
+option java_outer_classname = "CompositeProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Composite]
+// Composite Filter :ref:`configuration overview <config_http_filters_cors>`.
+// [#extension: envoy.filters.http.composite]
+
+// :ref:`Composite filter <config_http_filters_composite>` config. The composite filter config
+// allows delegating filter handling to an another filter as determined by a match result. This
+// makes it possible to use different filters or filter configurations based on the incoming request/response.
+//
+// This is intended to used with :ref:`ExtensionWithMatcher <envoy_api_msg_extensions.common.matching.v3.ExtensionWithMatcher>`
+// where a match tree is specified that indicates (via
+// :ref:`CompositeAction <envoy_api_msg_extensions.filters.http.composite.v3.CompositeAction>`) which filter configuration to
+// create and delegate to.
+//
+message Composite {
+}
+
+// Composite match action (see :ref:`matching docs <arch_overview_matching_api>` for more info on match actions).
+// This specifies the filter configuration of the filter that the composite filter should delegate filter interactions to.
+message CompositeAction {
```

Comment:
I might be inclined to call this `ExecuteFilterAction` to be more clear? It's in the composite namespace to it should be clear it's part of this filter?